src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+		self.pbSortSelectionIndex = index
+	end)
+	controls.itemSortSelection.tooltipFunc = function(tooltip)
+		tooltip:Clear()
+		tooltip:AddLine(16, "Weighted Sum searches ('?' button) will always sort")
+		tooltip:AddLine(16, "using descending weighted sum.")
+	end
+	controls.itemSortSelectionLabel = new("LabelControl",  {"TOPRIGHT",controls.itemSortSelection,"TOPRIGHT"}, -106, 0, 60, 18,  "^8Item Sort Selection:")
+	controls.fullPrice = new("EditControl", nil, -3, pane_height - 58, pane_width - 256, row_height, "", "Total Cost", "%Z")
+	top_pane_alignment_ref = {"TOPLEFT",controls.itemSetLabel,"TOPLEFT"}
+	for _, slotName in ipairs(baseSlots) do
+		local str_cnt = tostring(cnt)
+		self:PriceItemRowDisplay(controls, str_cnt, {name = slotName}, top_pane_alignment_ref, top_pane_alignment_width, top_pane_alignment_height, row_height)
+		top_pane_alignment_ref = {"TOPLEFT",controls['name'..str_cnt],"TOPLEFT"}
+		top_pane_alignment_width = 0
+		top_pane_alignment_height = 28
+		cnt = cnt + 1
+	end
+	local activeSocketList = { }
+	for nodeId, slot in pairs(self.itemsTab.sockets) do
+		if not slot.inactive then
+			t_insert(activeSocketList, nodeId)
+		end
+	end
+	table.sort(activeSocketList)
+	for _, nodeId in pairs(activeSocketList) do
+		local str_cnt = tostring(cnt)
+		self:PriceItemRowDisplay(controls, str_cnt, {name = self.itemsTab.sockets[nodeId].label, ref = nodeId}, top_pane_alignment_ref, top_pane_alignment_width, top_pane_alignment_height, row_height)
+		top_pane_alignment_ref = {"TOPLEFT",controls['name'..str_cnt],"TOPLEFT"}
+		top_pane_alignment_width = 0
+		top_pane_alignment_height = 28
+		cnt = cnt + 1
+	end
+	controls.close = new("ButtonControl", nil, 0, pane_height - 30, 90, row_height, "Done", function()
+        GlobalCache.useFullDPS = self.storedGlobalCacheDPSView
+		main:ClosePopup()
+	end)
+	controls.league = new("DropDownControl", {"TOPRIGHT",nil,"TOPRIGHT"}, -12, pane_height - 30, 100, 18, self.itemsTab.leagueDropList, function(index, value)
+		self.pbLeague = value.name
+		self.pbLeagueRealName = value.realname or value.name
+		self:SetCurrencyConversionButton(controls)
+	end)
+	controls.league.enabled = function()
+		return #self.itemsTab.leagueDropList > 1
+	end
+	controls.leagueLabel = new("LabelControl", {"TOPRIGHT",controls.league,"TOPLEFT"}, -4, 0, 20, 16, "League:")
+	controls.poesessidButton = new("ButtonControl", {"TOPLEFT",controls.leagueLabel,"TOPLEFT"}, -256, 0, 240, row_height, POESESSID ~= "" and "HAVE POESESSID" or colorCodes.WARNING.."NEED POESESSID", function()
+		local poesessid_controls = {}
+		poesessid_controls.sessionInput = new("EditControl", nil, 0, 18, 350, 18, #POESESSID == 32 and POESESSID or "<PASTE POESESSID FROM BROWSER>", "POESESSID", "%X", 32, function(buf)
+			if #poesessid_controls.sessionInput.buf == 32 then
+				POESESSID = poesessid_controls.sessionInput.buf
+				controls.poesessidButton.label = "HAVE POESESSID"
+			end
+		end)
```

Comment:
I was somehow able to break this by trying to import from an account that doesn't exist in the ImportTab, coming back to this modal, and it telling me I "HAVE POESESSID"

Reply:
FIXED